### ISSUE TITLE
Fatal error when Relationships are empty

### ISSUE THREAD

**Issue by alessandrofrancesconi** [tobscure/json-api#49]
I was following the example in the README regarding [Relationships](https://github.com/tobscure/json-api#relationships).

I'm implementing a very similar situation to 'comments', but with 'tags'. As you know, a post can have no tags at all, so there is a point where I pass an empty array to the `collection` method:

```
protected function tags()
    {
        return function ($post, $include, array $included, array $linked) {
            $serializer = new TagSerializer($included, $linked);
            
            $serializerTags = array();
            if ($include) $serializerTags = $post->tags;
            else foreach ($post->tags as $t) $serializerTags[] = $t->id;
            
            $tags = $serializer->collection($serializerTags); 
            $relationship = new Tobscure\JsonApi\Relationship($tags);

            return $relationship;
        };
    }
```
When `$serializerTags` is empty, `collection()` [returns NULL](https://github.com/tobscure/json-api/blob/b4cb8dff053bd9adc0bf471152c297eeb8bf242f/src/AbstractSerializer.php#L108)!

When passing a null parameter to the `Relationship` constructor, it throws:
`Catchable fatal error</b>:  Argument 1 passed to Tobscure\JsonApi\Relationship::__construct() must implement interface Tobscure\JsonApi\Elements\ElementInterface, null given`

So, what's the right way to handle relationships and empty cases?

**Comment by tobscure** [comment]
For now, you should be able to return null instead of a Relationship object if there is no relationship data. e.g.

```php
if (! count($serializerTags)) {
    return null;
}

$tags = $serializer->collection($serializerTags); 

return new Tobscure\JsonApi\Relationship($tags);
```

I can see that this isn't ideal though, because there still might be meta/links on the relationship even if there isn't any data. I reckon it should be a pretty easy fix, namely making the [Relationship constructor's $data argument](https://github.com/tobscure/json-api/blob/master/src/Relationship.php#L56) optional (i.e. `ElementInterface $data = null`). Don't have time to push/test right now though, but feel free to PR!

**Comment by tobscure** [comment]
This should be fixed/irrelevant in the upcoming 0.2 branch.
